packages/terra-badge/tests/nightwatch/badge-spec.js

```diff
@@ -1,20 +1,57 @@
+/* eslint-disable no-unused-expressions */
 // eslint-disable-next-line import/no-extraneous-dependencies
 const resizeTo = require('terra-toolkit/lib/nightwatch/responsive-helpers').resizeTo;
 
 module.exports = resizeTo(['tiny', 'small', 'medium', 'large', 'huge', 'enormous'], {
   'Displays a default badge with the provided text': (browser) => {
     browser.url(`${browser.launchUrl}/#/tests/badge-tests/default`);
+
+    browser.expect.element('#default-badge').to.be.present;
+    browser.expect.element('#default-badge span').text.to.equal('Default');
   },
 
-  'Displays intent badges with the provided text': (browser) => {
+  'Displays intent badges': (browser) => {
     browser.url(`${browser.launchUrl}/#/tests/badge-tests/intent`);
+
+    browser.expect.element('#default-badge').to.be.present;
+
+    browser.expect.element('#primary-badge').to.have.attribute('class').which.contains('primary');
+
+    browser.expect.element('#secondary-badge').to.have.attribute('class').which.contains('secondary');
+
+    browser.expect.element('#positive-badge').to.have.attribute('class').which.contains('positive');
+
+    browser.expect.element('#negative-badge').to.have.attribute('class').which.contains('negative');
+
+    browser.expect.element('#warning-badge').to.have.attribute('class').which.contains('warning');
+
+    browser.expect.element('#info-badge').to.have.attribute('class').which.contains('info');
   },
 
-  'Displays size badges with the provided text': (browser) => {
+  'Displays size badges': (browser) => {
     browser.url(`${browser.launchUrl}/#/tests/badge-tests/size`);
+
+    browser.expect.element('#tiny-badge').to.have.attribute('class').which.contains('tiny');
+
+    browser.expect.element('#small-badge').to.have.attribute('class').which.contains('small');
+
+    browser.expect.element('#medium-badge').to.have.attribute('class').which.contains('medium');
+
+    browser.expect.element('#large-badge').to.have.attribute('class').which.contains('large');
+
+    browser.expect.element('#huge-badge').to.have.attribute('class').which.contains('huge');
   },
 
-  'Displays icon badges with the provided text': (browser) => {
+  'Displays icon badges': (browser) => {
     browser.url(`${browser.launchUrl}/#/tests/badge-tests/icon`);
+
+    browser.expect.element('#text-right *:nth-child(1)').to.have.attribute('class').which.contains('terra-Icon');
```

Comment:
We could update this to 
`browser.expect.element('span[id="text-right"]:nth-child(1)')` so we don't have to use the * selector

Reply:
So, I had tried something similar before and again just now with your suggestion. I'm getting a failure because that selector is returning the parent and not the children.
`Expected element <span[id="text-right"]:nth-child(1)> to have attribute "class" which contains: "terra-Icon"  - expected "contains 'terra-Icon'" but got: "Badge__badge___33ckJ Badge__small___3uLPt Badge__default___3a0iu"`

You can see the same behavior in Chrome (Version 60.0.3112.113).
<img width="554" alt="screen shot 2017-09-13 at 1 55 10 pm" src="https://user-images.githubusercontent.com/3671289/30395378-902c51f8-988b-11e7-9a31-9603b27b96a7.png">

Is there something here I'm missing?

Reply:
Oh my bad. Yeah, I was wrong here. `span[id="text-right"]:nth-child(1)` will search the dom for all spans with the id of 'text-right', then since its specifying the first child, it will return the first instance. Sorry, bad suggestion here.